### PR TITLE
Test setup

### PR DESCRIPTION
**Pull request by khassel**
This is a first step to use electron only when needed in the tests, see #2408.
The idea comes from @fewieden see [this comment](https://github.com/MichMich/MagicMirror/issues/2649#issuecomment-917600122).

All tests using node-fetch and the translations test can run in server-only-mode instead using electron. So I make them work for this and moved all the remaining electron tests to a new category `electron` (from `e2e`) hoping we can shift more electron tests to `e2e` in the future.

I also changed `logger.js` again to be silent in the tests.

Tested all changes running mm in server-only mode and running with electron on a pi3.

Some dependencies were updated too.

### CODE REVIEW

File: tests/e2e/global-setup.js [hunk 1]
```diff
@@ -1,53 +1,16 @@
-/*
```
Comment (rejas) [inline tests/e2e/global-setup.js:1]: Dont remove the header, update it :-)

Comment (khassel) [reply tests/e2e/global-setup.js:1]: This is a new file, the old one still exists under `tests/electron`.

File: tests/electron/env_spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+const helpers = require("./global-setup");
```
Comment (rejas) [inline tests/electron/env_spec.js:1]: Here a header comment with license is missing

**Review by rejas — APPROVED** [review]

File: tests/electron/env_spec.js (hunk 2 above)
Comment (khassel) [reply tests/electron/env_spec.js:1]: this is missing in the most test files, I did split this file but the original also didn't contain a header comment.

**Pull request merged**